### DISCUSSION TITLE
Signed Fields

### DISCUSSION THREAD

**Discussion by AlexLao512** [SystemRDL/systemrdl-compiler#197] [General]
I was looking through the SystemRDL 2.0 specification, does anyone have an idea how to specify that an arbitrary width field (say, 5 bits) is a signed value so sign extension can either be done automatically in generated code or something?

**Comment by amykyta3** [comment]
Unfortunately the RDL spec doesn't define anything for signed fields.
In fact, the spec actually explicitly states that all integers are treated as unsigned (See 4.6).

That said, I suppose it would be possible to extend support for signed fields with a user-defined property.

**Reply by AlexLao512** [reply]
Yea, this is approximately what I concluded as well, just wasn't sure if I missed something in the specifications.
